src/gc/gcinterface.ee.h

```diff
@@ -128,6 +128,19 @@ class IGCToCLR {
     // barrier if it needs to be updated.
     virtual
     void StompWriteBarrier(WriteBarrierParameters* args) = 0;
+
+    // Signals to the finalizer thread that there are objects ready to
+    // be finalized.
+    virtual
+    void EnableFinalization() = 0;
+
+    // Returns true if there is additional work that the finalizer thread needs
+    // to do, outside of its duties of finalization.
+    virtual
+    bool HaveExtraWorkForFinalizer() = 0;
```

Comment:
The GC is not managing the finalizer thread, and so it is odd to have method on the GC/EE interface that returns whether there is more work for the finalizer thread. It also makes the interface more chatty than necessary, though it does not really matter in this case because of it is not a hot path.

Instead, I would introduce `bool` argument on the EnableFinalization method. The VM implementation of this method would be:

```
void EnableFinalization(bool foundFinalizers)
{
     if (foundFinalizers || HaveExtraWorkForFinalizer()) 
        hEventFinalizer->Set();
}
```